File: LeetCode/53.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        long long curr_sum = nums[0];
        long long ret = nums[0];
        int n = nums.size();
        for(int i=1; i<n; ++i){
            if(curr_sum + nums[i] < nums[i]) {
                curr_sum = nums[i];
            }
            else{
                curr_sum = curr_sum + nums[i];
            }
            ret = max(ret, curr_sum);
        }

        return ret;
    }
};

File: Coding Test/53.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        if len(nums) == 1:
            return nums[0]
        
        current_sum = -2
        best_sum = nums[0]
        for i, num in enumerate(nums, 1):
            current_sum = max(current_sum + num, num)
            best_sum =  max(best_sum, current_sum)
            
        return best_sum